torch/utils/cpp_extension.py

```diff
@@ -828,6 +828,39 @@ def CUDAExtension(name, sources, *args, **kwargs):
                 cmdclass={
                     'build_ext': BuildExtension
                 })
+
+    Compute capabilities:
+
+    By default the extension will be compiled to run on all archs of the cards visible during the
+    building process of the extension. If down the road a new card is installed the extension may
+    need to be recompiled.
+
+    This default may also clamp a higher arch to a lower one that has the same binary compatibility
+    if pytorch was compiled with that lower one. e.g. ``sm_86`` could be clamped down to `sm_80`.
+    For a variety of technical reasons the distributed pytorch binary doesn't build against the full
+    range of computer capabilities, e.g. it includes only `sm_60` and `sm_70`,but not `sm_61` and
+    `sm_75`. The not included ones are binary compatible with the included ones, but you might not
+    be getting the best performance.
+
+    However, you can explicitly specify which archs you want the extension to support like so:
+
+    TORCH_CUDA_ARCH_LIST="6.1 8.6" python build_my_extension.py
+    TORCH_CUDA_ARCH_LIST="5.2 6.0 6.1 7.0 7.5 8.0 8.6+PTX" python build_my_extension.py
+
+    The `+PTX` option is special and if provided as shown in the last example will support any card
+    whose compute capability was not compiled for and it'll use JIT at runtime instead (that's, of
+    course, if the instruction sets match - some old cards won't be possible to use.)
```

Comment:
```
   The `+PTX` option causes extension kernel binaries to include PTX instructions for the specified compute capability (CC).  PTX enables kernels to runtime-compile for any CC >= the specified compute CC (for example, `8.6+PTX` generates PTX that can runtime-compile for any GPU with CC >= 8.6).  This improves your binary's forward compatibility.  However, relying on older PTX to provide forward compat by runtime-compiling for newer CCs can modestly reduce performance on those newer CCs.  If you know exact CC(s) of the GPUs you want to target, you're always better off specifying them individually.  For example, if you want your extension to run on 8.0 and 8.6, `8.0+PTX` would work functionally because it includes PTX that can runtime-compile for both 8.0 and 8.6, but `8.0 8.6` would be better.
```

Reply:
Amazing! This is so much more clear - I will update it. Thank you, Michael

Reply:
`8.6+PTX` does generate ptx that can runtime compile for 8.6.  So I think `>=` is the technically correct thing to say (the 8.6 ptx is there, even though it's unused because the lowest-level SASS for 8.6 is also built in).

Reply:
also edited, accidentally said "compute CC" in one spot